multidict/_multidict_c.c

```diff
@@ -0,0 +1,1617 @@
+        "update",
+        (PyCFunction)multidict_update,
+        METH_VARARGS | METH_KEYWORDS,
+        multidict_update_doc
+    },
+    {
+        "__reduce__",
+        (PyCFunction)multidict_reduce,
+        METH_NOARGS,
+        NULL,
+    },
+    {
+        NULL,
+        NULL
+    }   /* sentinel */
+};
+
+static PyTypeObject multidict_type = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "multidict._multidict.MultiDict",                /* tp_name */
+    sizeof(MultiDictObject),                         /* tp_basicsize */
+    0,                                               /* tp_itemsize */
+    (destructor)multidict_tp_dealloc,                /* tp_dealloc */
+    0,                                               /* tp_print */
+    0,                                               /* tp_getattr */
+    0,                                               /* tp_setattr */
+    0,                                               /* tp_reserved */
+    (reprfunc)multidict_repr,                        /* tp_repr */
+    0,                                               /* tp_as_number */
+    &multidict_sequence,                             /* tp_as_sequence */
+    &multidict_mapping,                              /* tp_as_mapping */
+    0,                                               /* tp_hash */
+    0,                                               /* tp_call */
+    0,                                               /* tp_str */
+    0,                                               /* tp_getattro */
+    0,                                               /* tp_setattro */
+    0,                                               /* tp_as_buffer */
+    Py_TPFLAGS_DEFAULT
+        | Py_TPFLAGS_BASETYPE
+        | Py_TPFLAGS_HAVE_GC,                        /* tp_flags */
+    0,                                               /* tp_doc */
+    (traverseproc)multidict_tp_traverse,             /* tp_traverse */
+    (inquiry)multidict_tp_clear,                     /* tp_clear */
+    (richcmpfunc)multidict_tp_richcompare,           /* tp_richcompare */
+    0,                                               /* tp_weaklistoffset */
+    (getiterfunc)mulditict_tp_iter,                  /* tp_iter */
+    0,                                               /* tp_iternext */
+    multidict_methods,                               /* tp_methods */
+    0,                                               /* tp_members */
+    0,                                               /* tp_getset */
+    0,                                               /* tp_base */
+    0,                                               /* tp_dict */
+    0,                                               /* tp_descr_get */
+    0,                                               /* tp_descr_set */
+    0,                                               /* tp_dictoffset */
+    (initproc)multidict_tp_init,                     /* tp_init */
+    0,                                               /* tp_alloc */
```

Comment:
Python dict has the slot filled with `PyType_GenericAlloc`.
Can we do the same?

Reply:
Yes. Done.